File: src/GameEngine/MainMenu.cs
using System;
using System.Collections.Generic;

namespace GameEngine
{
    public class MainMenu
    {
        public List<string> MenuOptions { get; private set; }

        public MainMenu()
        {
            MenuOptions = new List<string>
            {
                "Start Game",
                "My Character",
                "Inventory",
                "Quests",
                "Community",
                "Market",
                "Settings",
                "Exit Game"
            };
        }

        public void Render()
        {
            Console.WriteLine("=== GOONZU MAIN MENU ===");
            for (int i = 0; i < MenuOptions.Count; i++)
            {
                Console.WriteLine($"{i + 1}. {MenuOptions[i]}");
            }
        }

        public void HandleInput(int input)
        {
            if (input < 1 || input > MenuOptions.Count)
            {
                Console.WriteLine("Invalid option.");
                return;
            }
            Console.WriteLine($"Selected: {MenuOptions[input - 1]}");
            // Add logic for each menu option
        }
    }
}

File: src/GameEngine/MainMenu.cpp
// MainMenu.cpp
#include "MainMenu.h"
#include <iostream>

MainMenu::MainMenu() {
    menuOptions = {
        "Start Game",
        "My Character",
        "Inventory",
        "Quests",
        "Community",
        "Market",
        "Settings",
        "Exit Game"
    };
}

void MainMenu::Render() {
    std::cout << "=== GOONZU MAIN MENU ===" << std::endl;
    for (size_t i = 0; i < menuOptions.size(); ++i) {
        std::cout << i + 1 << ". " << menuOptions[i] << std::endl;
    }
}

void MainMenu::HandleInput(int input) {
    if (input < 1 || input > menuOptions.size()) {
        std::cout << "Invalid option." << std::endl;
        return;
    }
    std::cout << "Selected: " << menuOptions[input - 1] << std::endl;
    // Add logic for each menu option
}

std::vector<std::string> MainMenu::GetMenuOptions() const {
    return menuOptions;
}
